### PR TITLE
Add visualiation of administrative units on all spatial resolutions

### LINKED ISSUE #165: Add overview of administrative regions to the documentation
This is to have some more images in the text. Being able to see all regions would be helpful.

Questions:
* How to handle the license of NUTS and GADM?
* Shall we show transmission links? If yes, which? Showing the all-neighbours-connected links may scare people off.

### PR DESCRIPTION
**Pull request by timtroendle**
Fixes #165  .

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [x] Minimal workflow tests pass
- [x] Tests added to cover contribution
- [x] Documentation updated
- [x] Configuration schema updated

### CODE REVIEW

File: scripts/docs/spatial_scope_and_resolutions.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    plt.rcParams.update({'font.size': 6})
+    fig = plt.figure(figsize=(4, 1.4))
+    axes = fig.subplots(1, 4, gridspec_kw={'width_ratios': [0.15, 0.35, 0.35, 0.15]})
+    for ax in axes:
+        ax.set_xticks([])
+        ax.set_yticks([])
+    sns.despine(fig, top=True, bottom=True, left=True, right=True)
+
+    # plot
+    _plot_text(national, regional, ax=axes[1])
```
[the remark this answers is not in the document]
Comment (timtroendle) [reply scripts/docs/spatial_scope_and_resolutions.py:34]: I tried it and I'd conclude that both work. The current version looks more like a box (or slide) for the reasons you mentioned while the flipped version gives a centralised impression: both text and map jitter on the extremes. I personally like the box/slide version better.

File: scripts/docs/spatial_scope_and_resolutions.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    _plot_map(regional, label=None, ax=axes[2], edge_width=0.1)
+    _plot_map(national, label=None, ax=axes[2], face_color="none", edge_width=0.3)
+
+    # adjust figure
+    fig.subplots_adjust(wspace=0, hspace=0, top=1, bottom=0, left=0, right=1)
+    fig.savefig(path_to_output, dpi=dpi, transparent=True)
```
[the remark this answers is not in the document]
Comment (timtroendle) [reply scripts/docs/spatial_scope_and_resolutions.py:40]: When I add this with everything else equal, the white borders get larger and the content smaller which is likely not what we want. I've never fully understood all that, but I think `bbox_inches` doesn't do anything else than trying to optimise the `subplots_adjust` settings, correct? I've set them manually in line 39 and would keep it that way.

File: scripts/docs/spatial_scope_and_resolutions.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    fig = plt.figure(figsize=(4, 1.4))
+    axes = fig.subplots(1, 4, gridspec_kw={'width_ratios': [0.15, 0.35, 0.35, 0.15]})
+    for ax in axes:
+        ax.set_xticks([])
+        ax.set_yticks([])
+    sns.despine(fig, top=True, bottom=True, left=True, right=True)
+
+    # plot
+    _plot_text(national, regional, ax=axes[1])
+    _plot_map(regional, label=None, ax=axes[2], edge_width=0.1)
```
[the remark this answers is not in the document]
Comment (timtroendle) [reply scripts/docs/spatial_scope_and_resolutions.py:35]: I've tried it and now I remember why I went for white: some coast lines (e.g. Norway, Greece, Croatia) are super complex and become weirdly bold when outlined. That's why I went for white: it's indistinguishable from the background. Personally, I don't consider it to be a disadvantage to use the same colour.

**Comment by timtroendle** [comment]
@brynpickering, I know you've approved already. But given that I did not address all of your comments and that I want to avoid to update this figure often (like in a follow-up PR addressing some of your remaining comments), I would like you to re-review.

File: scripts/vis/spatial_scope_and_resolutions.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    regional = gpd.read_file(path_to_regional_units).to_crs(EPSG3035).simplify(10000)
+    national = gpd.read_file(path_to_national_units).to_crs(EPSG3035).simplify(10000)
+
+    # prepare figure
+    plt.rcParams.update({'font.size': 6})
+    fig = plt.figure(figsize=(4, 1.4))
+    axes = fig.subplots(1, 4, gridspec_kw={'width_ratios': [0.15, 0.35, 0.35, 0.15]})
+    for ax in axes:
+        ax.axis("off")
+    sns.despine(fig, top=True, bottom=True, left=True, right=True)
```
Comment (brynpickering) [inline scripts/vis/spatial_scope_and_resolutions.py:31]: You shouldn't have to despine if you have `ax.axis("off")`.

File: scripts/vis/spatial_scope_and_resolutions.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
+from textwrap import dedent
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from eurocalliopelib.geo import EPSG3035
+
+BLUE = "#4F6DB8"
+LIGHT_BLUE = sns.light_palette(BLUE, as_cmap=False)[2]
```
Comment (brynpickering) [inline scripts/vis/spatial_scope_and_resolutions.py:10]: update this to `#aebcde` (output of `sns.light_palette(BLUE, as_cmap=False).as_hex()[2]`) and then you can remove the seaborn dependency.

File: scripts/vis/spatial_scope_and_resolutions.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    ax.set_aspect('equal')
+    data.plot(
+        ax=ax,
+        facecolor=face_color,
+        edgecolor=edge_color,
+        linewidth=edge_width
+    )
+    ax.set_xlim(MAP_MIN_X, MAP_MAX_X)
+    ax.set_ylim(MAP_MIN_Y, MAP_MAX_Y)
+    sns.despine(top=True, bottom=True, left=True, right=True)
```
Comment (brynpickering) [inline scripts/vis/spatial_scope_and_resolutions.py:82]: also not needed if you just use `ax.axis("off")`. Then seaborn isn't needed as a dependency

**Review by brynpickering — APPROVED** [review]
I'm OK with your final decision, thanks for checking the options. I've made comments to allow seaborn to be easily removed as a dependency.

**Pull request merged**